JitsiVideoSIPGWSession.js

```diff
@@ -0,0 +1,87 @@
+import EventEmitter from 'events';
+import { sendJibriIQ } from './modules/xmpp/VideoSIPGW';
+
+/**
+ * The event name for current sip video session state changed.
+ * @type {string} event name for sip video session state changed.
+ */
+const STATE_CHANGED = 'STATE_CHANGED';
+
+/**
+ * Jitsi video SIP GW session. Holding its state and able to start/stop it.
+ */
+export default class JitsiVideoSIPGWSession {
+
+    /**
+     * Creates new session with the desired sip address and display name.
+     *
+     * @param sipAddress - The sip address to use when starting the session.
+     * @param displayName - The display name to use for that participant.
+     * @param chatRoom - The chat room this session is bound to.
+     */
+    constructor(sipAddress, displayName, chatRoom) {
+        this.sipAddress = sipAddress;
+        this.displayName = displayName;
+        this.chatRoom = chatRoom;
+
+        this.eventEmitter = new EventEmitter();
+    }
+
+    /**
+     * Currently not implemented.
+     */
+    stop() {
+        sendJibriIQ(this, 'stop', this.chatRoom, null);
+    }
+
+    /**
+     * Starts a new session. Sends an iq to the focus.
+     */
+    start() {
+        sendJibriIQ(
```

Comment:
Both start and stop method do not seem to care about current state and allow to send IQ in possibly wrong state instead of failing immediately (it doesn't make sense to send stop when the session has been stopped already or has never been started).